tests/Integration/Behaviour/Features/Context/Domain/AliasFeatureContext.php

```diff
@@ -0,0 +1,142 @@
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain;
+
+use Behat\Gherkin\Node\TableNode;
+use Exception;
+use PHPUnit\Framework\Assert;
+use PrestaShop\PrestaShop\Core\Domain\Alias\Command\AddAliasCommand;
+use PrestaShop\PrestaShop\Core\Domain\Alias\Exception\AliasConstraintException;
+use PrestaShop\PrestaShop\Core\Grid\Query\AliasQueryBuilder;
+use PrestaShop\PrestaShop\Core\Search\Filters;
+use Tests\Integration\Behaviour\Features\Context\Util\PrimitiveUtils;
+use Tests\Resources\DatabaseDump;
+
+class AliasFeatureContext extends AbstractDomainFeatureContext
+{
+    /**
+     * @When I add alias with following information:
+     *
+     * @param TableNode $table
+     */
+    public function addAlias(TableNode $table): void
+    {
+        $data = $table->getRowsHash();
+
+        try {
+            $this->getCommandBus()->handle(new AddAliasCommand(
+                PrimitiveUtils::castStringArrayIntoArray($data['alias']),
+                $data['search']
+            ));
+        } catch (Exception $exception) {
+            $this->setLastException($exception);
+        }
+    }
+
+    /**
+     * @Then following aliases should exist:
+     *
+     * @param TableNode $table
+     */
+    public function assertAliasesList(TableNode $table): void
+    {
+        $expectedAliases = $table->getColumnsHash();
+
+        /** @var AliasQueryBuilder $aliasQueryBuilder */
+        $aliasQueryBuilder = $this->getContainer()->get(AliasQueryBuilder::class);
+        $aliases = $aliasQueryBuilder
+            ->getSearchQueryBuilder(new Filters([
+                'limit' => null,
+                'offset' => 0,
+                'orderBy' => 'id_alias',
+                'sortOrder' => 'asc',
+                'filters' => [],
+            ]))
+            ->execute()
+            ->fetchAllAssociative();
+
+        Assert::assertCount(count($expectedAliases), $aliases, 'Unexpected aliases count');
+
+        $idsByIdReferences = $this->assertAliasProperties($expectedAliases, $aliases);
+
+        foreach ($idsByIdReferences as $reference => $id) {
+            $this->getSharedStorage()->set($reference, $id);
+        }
+    }
+
+    /**
+     * @Then I should get error that alias constraint is invalid
+     *
+     * @return void
+     */
+    public function assertLastErrorIsInvalidAliasConstraint(): void
+    {
+        $this->assertLastErrorIs(AliasConstraintException::class);
+    }
```

Comment:
Following comments in feature file, this will have to assert Core/InvalidArgumentException and probably will fit both cases - when alias is empty and when search term is empty:

```suggestion
     * @Then I should get error that alias cannot be empty
     * @Then I should get error that search term cannot be empty
     *
     * @return void
     */
    public function assertLastErrorIsInvalidAliasConstraint(): void
    {
        $this->assertLastErrorIs(InvalidArgumentException::class);
    }
```